### ISSUE TITLE
Weird error messages if trying to make a static get_name() function 

### ISSUE THREAD

**Issue by miv391** [godotengine/godot#84326]
### Godot version

v4.2.beta3.mono.official [e8d57afae]

### System information

Godot v4.2.beta3.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 6GB (NVIDIA; 31.0.15.3640) - Intel(R) Core(TM) i5-3570K CPU @ 3.40GHz (4 Threads)

### Issue description

`foobar.gd`:

```gdscript
class_name Foobar
extends Object

static func get_name(i: int) -> String:
	return str(i)
```

`node_2d.gd` which uses `Foobar:`
```gdscript
extends Node2D

func _ready():
	# Invalid call to function 'get_name' in base 'GDScript'. Expected 0 arguments.
	var i = Foobar.get_name(42)

	# Parser Error: Too few arguments for "get_name()" call. Expected at least 1 but received 0.
	var j = Foobar.get_name()
```

If I try to call `Foobar`'s static `get_name` with one parameter (as it is implemented), I get an error message saying that 0 arguments are expected. But when I remove the argument, now the error message says that 1 argument is expected.

I'm guessing that `get_name` is a name that just cannot be used as a static function name. As a member function name it is usable.

The problem here is that the current error messages do not help to understand what is going on. If a static `get_name` is not allowed at all, I would expect an error message saying just that at `foobar.gd` where I try to implement that function.

### Steps to reproduce

As above.

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
That's because `get_name` is a method in `GDScript`, you can't name a static method the same as an existing one AFAIK

**Comment by miv391** [comment]
`get_name` still seems to be different than for example `sqrt`. This works as expected and can be used without error: 

`foobar.gd`:
```gdscript
static func sqrt(f: float) -> float:
	return 0.0
```
`node_2d.gd`:
```gdscript
print(Foobar.sqrt(1.0))
```
Also, the documentation doesn't seem to include that GDScript level `get_name` function which makes it a bit difficult to find why your code doesn't work.

**Comment by AThousandShips** [comment]
No, `get_name` is a method of `Resource` which `GDScript` inherits

**Comment by miv391** [comment]
Oh yes, `Resource` seems to have a `get_name` getter. A specific error message for these situations would be nice.

**Comment by AThousandShips** [comment]
There should be one already but might be mistaken, will investigate when I have time

**Comment by dalexeev** [comment]
* Duplicate of / related to #75392.
